File: backend/src/main/java/com/tunisia/commerce/service/NotificationService.java
package com.tunisia.commerce.service;

import com.tunisia.commerce.dto.notification.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface NotificationService {

    /**
     * Créer une notification pour l'ajout de produit par un importateur
     */
    NotificationResponseDTO createProductAdditionNotification(ProductAdditionNotificationDTO productAdditionDTO);

    /**
     * Créer une notification générique
     */
    NotificationResponseDTO createNotification(NotificationRequestDTO requestDTO);

    /**
     * Récupérer toutes les notifications d'un utilisateur
     */
    Page<NotificationResponseDTO> getUserNotifications(Long userId, Pageable pageable);

    /**
     * Récupérer les notifications non lues d'un utilisateur
     */
    List<NotificationResponseDTO> getUnreadNotifications(Long userId);

    /**
     * Récupérer le nombre de notifications non lues
     */
    long getUnreadCount(Long userId);

    /**
     * Marquer une notification comme lue
     */
    void markAsRead(Long notificationId);

    /**
     * Marquer toutes les notifications d'un utilisateur comme lues
     */
    void markAllAsRead(Long userId);

    /**
     * Traiter une action sur une notification (accepter/rejeter)
     */
    void handleNotificationAction(NotificationActionDTO actionDTO);

    /**
     * Supprimer une notification
     */
    void deleteNotification(Long notificationId);

    /**
     * Archiver les anciennes notifications
     */
    void archiveOldNotifications(int daysOld);

    /**
     * Envoyer les notifications par email (job programmé)
     */
    void sendPendingEmailNotifications();

    /**
     * Récupérer une notification par son ID
     */
    NotificationResponseDTO getNotificationById(Long notificationId);

    /**
     * Récupérer les notifications par entité cible
     */
    List<NotificationResponseDTO> getNotificationsByTargetEntity(String entityType, Long entityId);

    /**
     * Mettre à jour l'action d'une notification
     */
    void updateNotificationAction(Long notificationId, com.tunisia.commerce.enums.NotificationAction action);

    /**
     * Récupérer les notifications pour un produit spécifique
     */
    List<NotificationResponseDTO> getNotificationsByProduct(Long productId);
}

File: frontend/services/notificationService.ts
import axios from 'axios';
import { ProductAdditionData } from '../types/ProductAdditionData';
import { NotificationData } from '../types/NotificationData';
import { NotificationActionData } from '../types/NotificationActionData';

const API_BASE_URL = '/api';

const getAuthToken = () => {
  return localStorage.getItem('token') || sessionStorage.getItem('token') || '';
};

export const notificationService = {
  // Créer une notification pour l'ajout de produit
  createProductAdditionNotification: async (data: ProductAdditionData): Promise<NotificationData> => {
    const token = getAuthToken();
    const response = await axios.post(`${API_BASE_URL}/notifications/product-addition`, data, {
      headers: { 'Authorization': `Bearer ${token}` }
    });
    return response.data;
  },

  // Récupérer les notifications non lues d'un utilisateur
  getUnreadNotifications: async (userId: number): Promise<NotificationData[]> => {
    const token = getAuthToken();
    const response = await axios.get(`${API_BASE_URL}/notifications/user/${userId}/unread`, {
      headers: { 'Authorization': `Bearer ${token}` }
    });
    return response.data;
  },

  // Récupérer le nombre de notifications non lues
  getUnreadCount: async (userId: number): Promise<number> => {
    const token = getAuthToken();
    const response = await axios.get(`${API_BASE_URL}/notifications/user/${userId}/unread-count`, {
      headers: { 'Authorization': `Bearer ${token}` }
    });
    return response.data.count;
  },

  // Marquer une notification comme lue
  markAsRead: async (notificationId: number): Promise<void> => {
    const token = getAuthToken();
    await axios.put(`${API_BASE_URL}/notifications/${notificationId}/read`, {}, {
      headers: { 'Authorization': `Bearer ${token}` }
    });
  },

  // Traiter une action sur une notification
  handleNotificationAction: async (data: NotificationActionData): Promise<void> => {
    const token = getAuthToken();
    await axios.post(`${API_BASE_URL}/notifications/${data.notificationId}/action`, data, {
      headers: { 'Authorization': `Bearer ${token}` }
    });
  },

  // Récupérer toutes les notifications d'un utilisateur
  getUserNotifications: async (userId: number, page: number = 0, size: number = 20): Promise<NotificationData[]> => {
    const token = getAuthToken();
    const response = await axios.get(`${API_BASE_URL}/notifications/user/${userId}?page=${page}&size=${size}`, {
      headers: { 'Authorization': `Bearer ${token}` }
    });
    return response.data.content;
  }
};